server/README.md

```diff
@@ -254,8 +254,47 @@ cargo run --bin remote_server_cli -- initialise-from-central -u 'user1:password1
 cargo run --bin remote_server_cli -- refresh-dates
 ```
 
-## Logging
+# Logging
 
 By default, the server logs to console with a logging level of `Info`
 You can configure this, to log to a file, for example, with a rollover of log files based on file size.
-See the `example.yaml` file for the available options.
\ No newline at end of file
+See the `example.yaml` file for the available options.
+
+# Directory structure
+
+An overview of how files are organised in the server repo is shown below, with annotations as noted.
+The files have been split in this way, grouped in crates, partly to improve build times - as well as a logical organisation of grouped concerns.
+
+```
+server
+├─ android (hoisting for the android implementation)
+├─ cli
+├─ configuration (runtime configuration files)
+├─ data (contains reference data which can be used by the cli to initialise the database)
+├─ docker (a zola docker instance)
+├─ docs
+├─ graphql
+│  ├─ batch_mutations (batch mutations are grouped here)
```

Comment:
not sure if this should have more explanation - why batch mutations are here and not in the `mutations` under each table folder?

Reply:
hmm, here is evolution of our project structure:

We started with single crate project, it was partitioned by `areas`, repository, graphql, server, etc... Compilation times were very very slow, and it took awhile for rust analyzer to show error or help with auto completes etc.. 
At this point we broke down each `area` into it's own crate and used workspace, this allows for `incremental` compilation, and it helped a lot to speed up compile time, but was still slow for large crates with lots of derived macros like the graphql one. So we ended up breaking down graphql crate into seperate crates, one challenge here was that you are not allowed to have circular dependencies between crates, and we have circular dependencies in types (i.e. invoiceLine <-> invoice). 

I think batch mutations was split into it's own crate because it takes quite a bit of time to compile, and I think it also has combination of invoice and lines, thus we would need to consolidate those two crates.